internal/atomic/counter.go

```diff
@@ -0,0 +1,26 @@
+// Package atomic implements a simple threadsafe counter
+package atomic
+
+import (
+	"sync/atomic"
+)
+
+// Counter is a threadsafe integer counter.
+type Counter struct {
```

Comment:
Is this really needed? All the functions just directly call `atomic` and don't add any extra logic on top?

Reply:
Yeah, this is overkill - removed this package entirely and updated the LB to just use a `uint64` field + atomic directly.